spec/ios_get_app_version_spec.rb

```diff
@@ -0,0 +1,41 @@
+require 'spec_helper'
+
+describe Fastlane::Actions::IosGetAppVersionAction do
+  describe 'getting the public app version from the provided .xcconfig file' do
+    it 'parses the xcconfig file format correctly and gets the public version' do
+      xcconfig_mock_content = <<~CONTENT
+        // a comment
+        VERSION_SHORT=6
+        VERSION_LONG=6.30.0
+      CONTENT
+
+      allow(File).to receive(:exist?).and_return(true)
+
+      expect_version(xcconfig_mock_content: xcconfig_mock_content, expected_version: '6.30')
+    end
+
+    it 'parses the xcconfig file format correctly and gets the public hotfix version' do
+      xcconfig_mock_content = <<~CONTENT
+        VERSION_SHORT=6
+        // a comment
+        VERSION_LONG=6.30.1
+      CONTENT
+
+      allow(File).to receive(:exist?).and_return(true)
+
+      expect_version(xcconfig_mock_content: xcconfig_mock_content, expected_version: '6.30.1')
+    end
+
+    def expect_version(xcconfig_mock_content:, expected_version:)
+      xcconfig_mock_file_path = File.join('mock', 'file', 'path')
+
+      allow(File).to receive(:open).with(xcconfig_mock_file_path, 'r').and_yield(StringIO.new(xcconfig_mock_content))
```

Comment:
I really like the use of `StringIO` here to mock the `File.open` with a simple IO object; but that also means that we assume implementation details (i.e. that the action will use `File.open` to read the `.xcconfig` file, as opposed to any other mean to read that file, like `File.read` or using the `xcodeproj` gem and `Xcodeproj::Config.new` (which, tbh, is what `read_from_config_file` should really have been using all that time… ohhh legacy code…)

As such, I'd suggest to try not to rely on the knowledge of implementation details of how the action reads the file, and instead create the temp file in a temp folder (you have some helper methods to help you with that in `spec_helper.rb`) so that the test will still work even if we update the implementation details of how the `.xcconfig` file is read in the future.

Reply:
Ah, I totally agree, with this there's a big assumption on implementation details 👍 
I'll change it, thanks for the suggestion!

Reply:
Also, if you agree, I'll make this change on the [improvement PR](https://github.com/iangmaia/release-toolkit/pull/1) I made on top of this, using `Xcodeproj::Config`.

Reply:
Works for me 👍

Reply:
Done [here](https://github.com/iangmaia/release-toolkit/pull/1/commits/02a99f3f822de360db6587ce97ecf19fb25b2f0e)